### ISSUE TITLE
Help Wanted For STNet Training Process

### ISSUE THREAD

**Issue by Aria-iu** [ags3927/MERGE#2]
Hello author, while reproducing your code, I used the STNet dataset. I found that the PCC value on the validation set is around 0.3, which is significantly different from your results. I did not convert the jpg images to tiff files. At 15 epochs, the results were not ideal. I increased it to 50 epochs, and the results were somewhat better than at 15 epochs.

![Image](https://github.com/user-attachments/assets/5a470674-5998-4ee0-a9d7-82b694b802e5)

What can I do to achieve a PCC value around 0.5?

**Comment by Aria-iu** [comment]
As mentioned in issue 1, can ResNet connected with several linear layers achieve such a high PCC value (0.5)? Do I need to increase the learning rate and epochs? My current learning rate is 0.00005, and the number of epochs is 50.

**Comment by ags3927** [comment]
Hi, for the CNN training, are you using the same parameters as mentioned in the documentation or in issue 1 by the other user? (Except epochs of course, you said you increased it to 50, which I don't think you need to do)

**Comment by Aria-iu** [comment]
Thank you for your response. Here is my configuration for training the CNN.

![Image](https://github.com/user-attachments/assets/7f77392b-e5e3-4f62-a4c0-313c5550c0a8)

**Comment by Aria-iu** [comment]
I made some comparisons of the hyperparameters, and here are my training results.

![Image](https://github.com/user-attachments/assets/af359e21-a6df-4ad2-9dd7-60824a5ae815)

**Comment by ags3927** [comment]
I am re-running my experiments with your exact parameter sets and reporting back in a few hours. In the mean time, can you re-run with a larger batch size? Maybe 16 or 32?

**Comment by Aria-iu** [comment]
Ok，I will increase the batch size to 16 on STNet DataSet.

**Comment by Aria-iu** [comment]
Due to equipment reasons, my training process only has data from the first 4 folds.Increasing the batch_size can improve the accuracy of the model, but the improvement is not large under the condition of epoch=15.

![Image](https://github.com/user-attachments/assets/0d58f702-3309-4f0a-b665-baa9e4da73f3)

**Comment by ags3927** [comment]
I think I have figured out the reason for the inconsistency. In my original code I set shuffle=True for the CNN dataloader, but in the organized repository I uploaded here, I set it to False. I think the model does not learn to generalize properly unless we shuffle the data during each iteration. Please confirm if this is indeed the reason. 

Could you re-run a few of the tests after setting shuffle=True in the CNN dataloader? Code location is [here](https://github.com/ags3927/MERGE/blob/b643cf20de351f531fcf77557f9ebc7d7712f6af/utils/data.py#L201C118-L201C123).

For reference, these are the configs I am using (GNN configs are irrelevant during the CNN training, so I skipped them to avoid clutter):

```
General:
  seed: 3927

Data:
  dataset: stnet
  num_genes: 250
  folds: 8
  path: /data/ani/repos/merge_main/merge/data/stnet
  slides: /data/ani/repos/merge_main/merge/config/slides_stnet.csv

CNN:
  pretrained_path: 'pretrained/model-low-v1.pth'
  batch_size: 32
  epochs: 15
  dropout: 0.2
  optimizer:
    type: adam
    lr: 0.00005
    weight_decay: 0.0
  scheduler:
    type: step
    step_size: 2
    gamma: 0.5
```

**Comment by ags3927** [comment]
If you can confirm that this was indeed the reason for the poor performance, I will make a commit and fix the main repo. Thanks.

**Comment by Aria-iu** [comment]
Thank you. I'll try this modification as soon as possible.

**Comment by Aria-iu** [comment]
Hi author,It seems that this is indeed the issue.

<img width="989" alt="Image" src="https://github.com/user-attachments/assets/b0300d6c-2db0-4403-98e3-93b8858a256a" />

**Comment by ags3927** [comment]
Fixed in commit 75c6a3b259dc18ed6c890b51435e8c4eec5253a6. Please close when convenient @Aria-iu 😄 .
